sdk/python/kfp/_client.py

```diff
@@ -255,8 +255,19 @@ def _load_context_setting_or_default(self):
         self._context_setting = json.load(f)
     else:
       self._context_setting = {
-        'namespace': '',
+        'namespace': self._guess_user_namespace_or_default(),
       }
+
+  def _guess_user_namespace_or_default(self):
+    """Guess user namespace from pod service account metadata"""
+    SA_NAMESPACE_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/namespace'
+    default_namespace = ''
+
+    if os.path.isfile(SA_NAMESPACE_PATH):
```

Comment:
Sorry, I am not very familiar with python. Will this ever throw an exception? For example, no permission

Reply:
No, it won't throw exception based on current implementation: https://github.com/python/cpython/blob/master/Lib/genericpath.py#L27-L33

E.g.:

```Python
$ ls /root/
ls: cannot open directory /root/: Permission denied
$ python
Python 2.7.5 (default, Aug  7 2019, 00:51:29) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> st = os.stat('/root/anyfile')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 13] Permission denied: '/root/file'
>>> os.path.isfile('/root/anyfile')
False
```